File: privacystreams-app/src/main/java/io/github/privacystreams/app/db/PStreamContract.java
package io.github.privacystreams.app.db;

import android.provider.BaseColumns;

import io.github.privacystreams.core.Item;
import io.github.privacystreams.location.Geolocation;

public final class PStreamContract {
    private PStreamContract() {}

    public static class BaseEntry implements BaseColumns {
        public static final String TIME_CREATED = Item.TIME_CREATED;
    }

    public static class GeolocationEntry extends BaseEntry {
        public static final String BEARING = Geolocation.BEARING;
        public static final String ACCURACY = Geolocation.ACCURACY;
        public static final String SPEED = Geolocation.SPEED;
        public static final String PROVIDER = Geolocation.PROVIDER;
        public static final String LATITUDE = "latitude";
        public static final String LONGITUDE = "longitude";
    }
}

File: privacystreams-app/src/main/java/io/github/privacystreams/app/db/PStreamContract.kt
package io.github.privacystreams.app.db

import android.provider.BaseColumns

import io.github.privacystreams.core.Item
import io.github.privacystreams.location.Geolocation
import io.github.privacystreams.notification.Notification

object PStreamContract {

    class GeolocationEntry : BaseColumns {
        companion object {
            val TIME_CREATED = Geolocation.TIME_CREATED     // Long
            val TIMESTAMP = Geolocation.TIMESTAMP           // Long
            val BEARING = Geolocation.BEARING               // Float
            val ACCURACY = Geolocation.ACCURACY             // Float
            val SPEED = Geolocation.SPEED                   // Float
            val PROVIDER = Geolocation.PROVIDER             // String
            val LATITUDE = "latitude"                       // Double
            val LONGITUDE = "longitude"                     // Double
        }
    }

    class NotificationEntry : BaseColumns {
        companion object {
            val TIME_CREATED = Notification.TIME_CREATED    // Long
            val POST_TIME = Notification.POST_TIME          // Long
            val ACTION = Notification.ACTION                // String
            val CATEGORY = "category"                       // String
            val PACKAGE_NAME = "package_name"               // String
            val TITLE = "title"                             // String
            val TEXT = "text"                               // String
        }
    }
}
